src/ape/managers/query.py

```diff
@@ -0,0 +1,79 @@
+from typing import TYPE_CHECKING, Dict, Optional
+
+import pandas as pd
+
+from ape.api import QueryAPI, QueryType
+from ape.exceptions import QueryEngineError
+from ape.utils import cached_property
+
+if TYPE_CHECKING:
+    from ape.managers.networks import NetworkManager
+    from ape.plugins import PluginManager
+
+
+class QueryManager:
+    """
+    A singelton that manages all query sources.
+
+    Args:
+        query (``QueryType``): query to execute
+
+    Usage example::
+
+        chain.blocks.query()
```

Comment:
Better example?
```suggestion
        chain.blocks.query("size")
```

Reply:
Yes, the syntax for a DataFrame is slightly different.
```python
biggest_block_size = chain.blocks.query("size").max()
```